Automattic/pocket-casts-android
# Patron - Add icons

## Description

This adds icons Patron icons and starts upsell flow to show only Patron plans.

## Testing Instructions

Test.1 Feature Flag disabled

1. Launch the app without being logged in
2. Go to Profile -> Settings -> Appearance
3. ✅ Notice there are no Patron app icons

Test.2 Feature Flag enabled - Plus only

1. Enable the feature flag for Patron
2. Login with a Plus account
3. Go to Profile -> Settings -> Appearance
4. Notice that Patron app icons are disabled
5. Tap on a Patron app icon
6. ✅  Notice that Upsell flow starts with only Patron plans

Test.3 Feature Flag enabled - Patron

1. Enable the feature flag and SignInState for Patron
2. Install the app
3. Login with Plus account
4. Go to Profile -> Settings -> Appearance
6. Notice that Patron app icons are enabled
7. Tap on a Patron icon
8. ✅  Notice that you can now change the icon

## Screenshots or Screencast 

Plus and Patron disabled | Plus enabled and Patron Disabled | Patron enabled
----|----|-----
![0 Screenshot_20230511_182005](https://github.com/Automattic/pocket-casts-android/assets/1405144/e6af9d3b-53ca-4e0f-a240-ccc07fe24774) | ![1 Screenshot_20230511_182051](https://github.com/Automattic/pocket-casts-android/assets/1405144/33791b7c-2d7b-4931-9259-626f0728e731) | ![2 Screenshot_20230511_182131](https://github.com/Automattic/pocket-casts-android/assets/1405144/60779b9b-77fe-4108-a8d9-d4b234694d85)


https://github.com/Automattic/pocket-casts-android/assets/1405144/e5496e13-02a2-45b5-867c-ffbaf68a4c43


## Checklist
- If this is a user-facing change, I have added an entry in CHANGELOG.md
- Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- for accessibility with TalkBack
